pipeline/pipeline.go

```diff
@@ -52,23 +52,45 @@ const (
 
 type finalizeFn = func(event *Event, notifyInput bool, backEvent bool)
 
+type MetricsController interface {
+	MetricsCounters
+	MetricsGauges
+}
+
+type MetricsCounters interface {
+	RegisterCounter(name, help string)
+	IncCounter(name string)
+	AddCounter(name string, value float64)
+}
+
+type MetricsGauges interface {
+	RegisterGauge(name, help string)
+	IncGauge(name string)
+	DecGauge(name string)
+	AddGauge(name string, value float64)
+	SetGauge(name string, value float64)
+}
+
```

Comment:
It will be better under main interfaces: InputPluginController , ActionPluginController , OutputPluginController